lottie/CMakeLists.txt

```diff
@@ -0,0 +1,34 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+add_library(lottie STATIC)
+
+find_package(Qt6 REQUIRED COMPONENTS Core Qml)
+target_link_libraries(lottie PRIVATE Qt6::Core Qt6::Qml)
+target_include_directories(lottie PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}/lib)
+
+target_sources(lottie PRIVATE
+    lib/lottie.cpp
+    lib/lottie.h
+    lib/lottieprivate.cpp
+    lib/lottieprivate.h
+    lib/lottieprivatedocument.cpp
+    lib/lottieprivatedocument.h
+    lib/lottieprivatenavigator.cpp
+    lib/lottieprivatenavigator.h
+    lib/lottieprivatewindow.cpp
+    lib/lottieprivatewindow.h
+    lib/lottiestatus.h
+    lib/lottie.qrc
+)
+
+## This writes into the source directory to generate lottie.mjs. Which is frowned-upon. :`(
+get_filename_component(MVPN_SCRIPT_DIR ${CMAKE_CURRENT_SOURCE_DIR}/../scripts ABSOLUTE)
+add_custom_command(
+    OUTPUT ${CMAKE_CURRENT_SOURCE_DIR}/lib/lottie/lottie.mjs
+    MAIN_DEPENDENCY lib/lottie/lottie_wrap.js.template
+    DEPENDS lib/lottie/lottie.min.js
+    WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/lib/lottie
+    COMMAND python3 ${MVPN_SCRIPT_DIR}/utils/make_template.py -f __LOTTIE__=lottie.min.js -o lottie.mjs lottie_wrap.js.template
```

Comment:
can we have the script into the lottie folder?

Reply:
Lottie does not have ref with the rest of the codebase. Can we maybe dup the script?